### PR TITLE
[i18n-ar] Translated file : `docs/source/ar/task_summary.md` into Arabic

### PR DESCRIPTION
**Pull request by AhmedAlmaghz**
## What does this PR do?
Translated the `docs/source/ar/task_summary.md` file of the documentation to `Arabic العربية`. I use transformers for my side-project and wanted to contribute. As a beginner, this seemed appropriate.

### Some questions:

- I can't access the pr documentation endpoint for `Arabic العربية`. Is there a missing step?
- I left all the other files alone after copying the English docs. Is this acceptable for future updates or should I add contents to the `_toctree.yml` only when translation is complete? Should I delete all other files for a cleaner review?
- In `Arabic`, people use both `Transformer` and `المحولات` to describe the model. However, as this is more of a product of HuggingFace I opted not to translate it. May you please let me know which you would prefer?

Thank you in advance for your review.

Part of #32435

### Before submitting

- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests), Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/) ?Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

 ### Who can review?
 @stevhliu , @MKhalusova , @abodacs, may you please review this PR?

### CODE REVIEW

**Comment by AhmedAlmaghz** [comment]
@abodacs

Next Review :
# CONCEPTUAL GUIDES

- [x]  Philosophy #33064
- [x]  Glossary #33038
- [ ]  What 🤗 Transformers can do #33073
- [ ]  How 🤗 Transformers solve tasks #33074
- [ ]  The Transformer model family #33047
- [ ]  Summary of the tokenizers #33078
- [ ]  Attention #33021
- [ ]  Padding and truncation #33050
- [ ]  BERTology #33024
- [ ]  Perplexity of fixed-length models #33063
- [ ]  Pipelines for webserver inference #33066
- [ ]  Model training anatomy #33045
- [ ] Getting the most out of LLMs #33043

File: docs/source/ar/task_summary.md [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+... )
+>>> print(
+...     f"score: {round(preds['score'], 4)}, start: {preds['start']}, end: {preds['end']}, answer: {preds['answer']}"
+... )
+score: 0.9327, start: 30, end: 54, answer: huggingface/transformers
+```
+
+### التلخيص
+
+ينشئ التلخيص نسخة أقصر من نص من نص أطول مع محاولة الحفاظ على معظم معنى المستند الأصلي. التلخيص هو مهمة تسلسل إلى تسلسل؛ فهو يخرج تسلسل نص أقصر من الإدخال. هناك الكثير من الوثائق الطويلة التي يمكن تلخيصها لمساعدة القراء على فهم النقاط الرئيسية بسرعة. مشاريع القوانين والوثائق القانونية والمالية وبراءات الاختراع والأوراق العلمية هي مجرد أمثلة قليلة للوثائق التي يمكن تلخيصها لتوفير وقت القراء وخدمة كمساعد للقراءة.
```
Comment (abodacs) [inline docs/source/ar/task_summary.md:226]:
```suggestion
ينشئ التلخيص (Summarization) نسخة مختصرة من نص طويل مع محاولة الحفاظ على معظم معنى النص الأصلي. التلخيص هو مهمة تسلسل إلى تسلسل(Sequence-to-Sequence)؛؛ فهو تُنتج تسلسلًا نصيًا أقصر من النص المُدخل. هناك الكثير من المستندات الطويلة التي يمكن تلخيصها لمساعدة القراء على فهم النقاط الرئيسية بسرعة. مشاريع القوانين والوثائق القانونية والمالية وبراءات الاختراع والأوراق العلمية هي مجرد أمثلة قليلة للوثائق التي يمكن تلخيصها لتوفير وقت القراء وخدمة كمساعد للقراءة.
```

File: docs/source/ar/task_summary.md [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+
+مثل الإجابة على الأسئلة، هناك نوعان من التلخيص:
+
+* الاستخراجية: تحديد واستخراج الجمل الأكثر أهمية من النص الأصلي
+* التلخيصي: إنشاء ملخص الهدف (الذي قد يتضمن كلمات جديدة غير موجودة في وثيقة الإدخال) من النص الأصلي؛ يستخدم النهج التلخيصي [`SummarizationPipeline`]
```
Comment (abodacs) [inline docs/source/ar/task_summary.md:231]:
```suggestion
* الاستخراجية (Extractive): تحديد واستخراج أهم الجمل من النص الأصلي
* التجريدي (Abstractive): إنشاء ملخص مستهدف (الذي قد يتضمن كلمات جديدة غير موجودة في النص الأصلي) انطلاقًا من النص الأصلي؛ يستخدم نهج التلخيص التجريدي [`SummarizationPipeline`]
```

File: docs/source/ar/task_summary.md [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+... )
+[{'summary_text': ' The Transformer is the first sequence transduction model based entirely on attention . It replaces the recurrent layers most commonly used in encoder-decoder architectures with multi-headed self-attention . For translation tasks, the Transformer can be trained significantly faster than architectures based on recurrent or convolutional layers .'}]
+```
+
+### الترجمة
+
+تترجم الترجمة تسلسل نص بلغة واحدة إلى لغة أخرى. من المهم مساعدة الأشخاص من خلفيات مختلفة على التواصل مع بعضهم البعض، ومساعدة المحتوى على الوصول إلى جمهور أوسع، وحتى أن يكون أداة تعليمية لمساعدة الأشخاص على تعلم لغة جديدة. إلى جانب التلخيص، تعد الترجمة مهمة تسلسل إلى تسلسل، مما يعني أن النموذج يتلقى تسلسل إدخال ويعيد تسلسل الإخراج المستهدف.
```
Comment (abodacs) [inline docs/source/ar/task_summary.md:245]:
```suggestion
تحوّل الترجمة تسلسل نص بلغة إلى لغة أخرى. من المهم مساعدة الأشخاص من خلفيات مختلفة على التواصل مع بعضهم البعض، ومساعدة المحتوى على الوصول إلى جمهور أوسع، وحتى أن يكون أداة تعليمية لمساعدة الأشخاص على تعلم لغة جديدة. إلى جانب التلخيص، تعد الترجمة مهمة من نوع تسلسل إلى تسلسل، حيث يتلقى النموذج تسلسلًا مُدخلًا ويُعيد تسلسلًا مُخرَجًا مُستهدفًا.
```

File: docs/source/ar/task_summary.md [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+
+>>> text = "translate English to French: Hugging Face is a community-based open-source platform for machine learning."
+>>> translator = pipeline(task="translation", model="google-t5/t5-small")
+>>> translator(text)
+[{'translation_text': "Hugging Face est une tribune communautaire de l'apprentissage des machines."}]
+```
+
+### نمذجة اللغة
+
+نمذجة اللغة هي مهمة التنبؤ بكلمة في تسلسل نص. لقد أصبح مهمة NLP شائعة للغاية لأن النموذج اللغوي المسبق التدريب يمكن أن يتم ضبطه بشكل دقيق للعديد من مهام التدفق السفلي الأخرى. في الآونة الأخيرة، كان هناك الكثير من الاهتمام بنماذج اللغة الكبيرة (LLMs) التي توضح التعلم الصفري أو القليل من التعلم. وهذا يعني أن النموذج يمكنه حل المهام التي لم يتم تدريبه عليها بشكل صريح! يمكن استخدام نماذج اللغة لإنشاء نص سلس ومقنع، على الرغم من أنه يجب أن تكون حذرًا لأن النص قد لا يكون دائمًا دقيقًا.
```
Comment (abodacs) [inline docs/source/ar/task_summary.md:260]:
```suggestion
نمذجة اللغة (Language Modeling) هي مهمة التنبؤ بالكلمة التالية في تسلسل نصي. لقد أصبح مهمة NLP شائعة للغاية لأن النموذج اللغوي المسبق التدريب يمكن أن يتم ضبطه بشكل دقيق للعديد من مهام الأخرى. في الآونة الأخيرة، كان هناك الكثير من الاهتمام بنماذج اللغة الكبيرة (LLMs) التي توضح التعلم من الصفر أو من عدد قليل من الأمثلة (Zero-shot or Few-shot Learning). وهذا يعني أن النموذج يمكنه حل المهام التي لم يتم تدريبه عليها بشكل صريح! يمكن استخدام نماذج اللغة لإنشاء نص سلس ومقنع، على الرغم من أنه يجب أن تكون حذرًا لأن النص قد لا يكون دائمًا دقيقًا.
```

File: docs/source/ar/task_summary.md [hunk 5]
```diff
@@ -0,0 +1,323 @@
...
+
+هناك نوعان من نمذجة اللغة:
+
+* السببية: هدف النموذج هو التنبؤ بالرمز التالي في تسلسل، ويتم إخفاء الرموز المستقبلية
```
Comment (abodacs) [inline docs/source/ar/task_summary.md:264]:
```suggestion
* السببية(Causal): هدف النموذج هو التنبؤ بالرمز (Token)  التالي في التسلسل، ويتم إخفاء الرموز المستقبلية (Masking).
```

File: docs/source/ar/task_summary.md [hunk 6]
```diff
@@ -0,0 +1,323 @@
...
+
+```py
+>>> from transformers import pipeline
+
+>>> prompt = "Hugging Face is a community-based open-source platform for machine learning."
+>>> generator = pipeline(task="text-generation")
+>>> generator(prompt)  # doctest: +SKIP
+```
+
+* الأقنعة: هدف النموذج هو التنبؤ برمز مقنع في تسلسل مع الوصول الكامل إلى الرموز في التسلسل
```
Comment (abodacs) [inline docs/source/ar/task_summary.md:274]:
```suggestion
* المقنّع (Masked): هدف النموذج هو التنبؤ برمز مُخفيّ ضمن التسلسل مع الوصول الكامل إلى الرموز  الأخرى في التسلسل
```

File: docs/source/ar/task_summary.md [hunk 7]
```diff
@@ -0,0 +1,323 @@
...
+>>> preds
+[{'score': 0.2236,
+  'token': 1761,
+  'token_str': ' platform',
+  'sequence': 'Hugging Face is a community-based open-source platform for machine learning.'}]
+```
+  
+## متعدد الوسائط:
+
+تتطلب المهام متعددة الوسائط من النموذج معالجة وسائط بيانات متعددة (نص أو صورة أو صوت أو فيديو) لحل مشكلة معينة. يعد إنشاء عنوان الصورة مثالاً على مهمة متعددة الوسائط حيث يأخذ النموذج صورة كإدخال وينتج تسلسل نصي يصف الصورة أو بعض خصائصها.
```
Comment (abodacs) [inline docs/source/ar/task_summary.md:298]:
```suggestion
تتطلب المهام متعددة الوسائط (Multimodal) من النموذج معالجة وسائط بيانات متعددة (نص أو صورة أو صوت أو فيديو) لحل مشكلة معينة. يعد وصف الصورة (Image Captioning) مثالاً على مهمة متعددة الوسائط حيث يأخذ النموذج صورة كمدخل وينتج تسلسل نصيًا يصف الصورة أو بعض خصائصها.
```

File: docs/source/ar/task_summary.md [hunk 8]
```diff
@@ -0,0 +1,323 @@
...
+
+على الرغم من أن النماذج متعددة الوسائط تعمل مع أنواع بيانات أو طرائق مختلفة، إلا أن خطوات المعالجة المسبقة تساعد النموذج داخليًا على تحويل جميع أنواع البيانات إلى تضمينات (متجهات أو قوائم من الأرقام التي تحتوي على معلومات ذات معنى حول البيانات). بالنسبة لمهمة مثل إنشاء عنوان للصورة، يتعلم النموذج العلاقات بين تضمينات الصور وتضمينات النص.
```
Comment (abodacs) [inline docs/source/ar/task_summary.md:300]:
```suggestion
على الرغم من أن النماذج متعددة الوسائط تعمل مع أنواع أو وسائط بيانات مختلفة، إلا أن خطوات المعالجة المسبقة تساعد النموذج داخليًا على تحويل جميع أنواع البيانات إلى متجهات تضمين (Embeddings) (متجهات أو قوائم من الأرقام التي تحتوي على معلومات ذات معنى حول البيانات). بالنسبة لمهمة مثل وصف الصورة، يتعلم النموذج العلاقات بين متجهات تضمين الصور ومتجهات تضمين النص.
```

File: docs/source/ar/task_summary.md [hunk 9]
```diff
@@ -0,0 +1,323 @@
...
+
+### الإجابة على أسئلة المستندات:
+
+الإجابة على أسئلة المستندات هي مهمة تقوم بالإجابة على أسئلة اللغة الطبيعية من مستند. على عكس مهمة الإجابة على الأسئلة على مستوى الرمز التي تأخذ النص كإدخال، فإن الإجابة على أسئلة المستندات تأخذ صورة لمستند كإدخال بالإضافة إلى سؤال حول المستند وتعيد الإجابة. يمكن استخدام الإجابة على أسئلة المستندات لتفسير المستندات المهيكلة واستخراج المعلومات الرئيسية منها. في المثال أدناه، يمكن استخراج المبلغ الإجمالي والتغيير المستحق من الإيصال.
```
Comment (abodacs) [inline docs/source/ar/task_summary.md:304]:
```suggestion
الإجابة على أسئلة المستندات  (Document Question Answering) هي مهمة تقوم بالإجابة على أسئلة اللغة الطبيعية من مستند مُعطى. على عكس مهمة الإجابة على الأسئلة على مستوى الرموز (Token-Level) التي تأخذ نصًا كمدخل، فإن الإجابة على أسئلة المستندات تأخذ صورة لمستند كمدخل بالإضافة إلى سؤال هذا حول المستند وتعيد الإجابة. يمكن استخدام الإجابة على أسئلة المستندات لتفسير المستندات المُنسّقة واستخراج المعلومات الرئيسية منها. في المثال أدناه، يمكن استخراج المبلغ الإجمالي والمبلغ المُسترد من إيصال الدفع..
```

**Review by abodacs — CHANGES_REQUESTED** [review]
سﻻم عليكم
جزاء الله خيرا
@AhmedAlmaghz 

تمت المراجعة

**Comment by AhmedAlmaghz** [comment]
Thank you @abodacs

File: docs/source/ar/task_summary.md [hunk 10]
```diff
@@ -0,0 +1,323 @@
...
+>>> from transformers import pipeline
+>>> from PIL import Image
+>>> import requests
+
+>>> url = "https://huggingface.co/datasets/hf-internal-testing/example-documents/resolve/main/jpeg_images/2.jpg"
+>>> image = Image.open(requests.get(url, stream=True).raw)
+
+>>> doc_question_answerer = pipeline("document-question-answering", model="magorshunov/layoutlm-invoices")
+>>> preds = doc_question_answerer(
+...     question="ما هو المبلغ الإجمالي؟",
```
Comment (abodacs) [inline docs/source/ar/task_summary.md:316]: @AhmedAlmaghz 
اعتقد ﻻ يجب ان نترجم السؤال هنا 
ﻷن المودل ﻻ يدعم العربيه على ما اعتقد

ﻷن إذا اخد شخص الكود copy/paste الناتج ن يكون دقيقا
ما رأيك؟

**Review by abodacs — APPROVED** [review]
@AhmedAlmaghz 
سﻻم عليكم
جزاء الله خيرا

File: docs/source/ar/task_summary.md (hunk 10 above)
Comment (DSYemen) [reply docs/source/ar/task_summary.md:316]: نعم.
لا يجب ان نترجم النصوص التي في الاكواد لأننا نستخدم أكود الأخرين وقد لا تعمل مع اللغة العربية , خاصة مغ نماذج الذكاء الاصطناعي لأن معضمها لا يدعم اللغة العربية .

Comment (AhmedAlmaghz) [reply docs/source/ar/task_summary.md:316]: كل ذلك صحيح
شكراً للتفاعل @DSYemen  @abodacs

**Review by abodacs — APPROVED** [review]
@AhmedAlmaghz 
برجاء معالجة التعارض

**Pull request closed without merging**